app/src/common/shared/com/igalia/wolvic/ui/widgets/TrayWidget.java

```diff
@@ -830,8 +831,12 @@ private void updateWifi() {
             WifiManager wifiManager = (WifiManager) getContext().getSystemService(Context.WIFI_SERVICE);
             if (wifiManager != null) {
                 WifiInfo wifiInfo = wifiManager.getConnectionInfo();
-                // TODO: Deprecated calculateSignalLevel(int, int), see https://github.com/Igalia/wolvic/issues/802
-                int level = WifiManager.calculateSignalLevel(wifiInfo.getRssi(), 4);
+                int level;
+                if (Build.VERSION.SDK_INT > Build.VERSION_CODES.Q) {
+                    level = wifiManager.calculateSignalLevel(wifiInfo.getRssi());
+                } else {
+                    level = WifiManager.calculateSignalLevel(wifiInfo.getRssi(), 4);
+                }
```

Comment:
Add "Fixes #802" to the commit message for traceability and automatic issue handling.